File: Assignment__15/Question4.c
#include<stdio.h>
#include<stdbool.h>

int Display(int iNo)
{
    int iDigit = 0;
    int iMulti = 1;

    if(iNo < 0)
    {
        iNo = -iNo;
    }

    while(iNo !=0)
    {
        iDigit = iNo % 10;  
        iMulti = iMulti * iDigit;                     
        iNo = iNo / 10;  
    }

    return iMulti;
}

int main()
{
    int iValue = 0;
    int iResult = 0;

    printf("Enter the number : \n");
    scanf("%d", &iValue);

    iResult = Display(iValue);

    printf("The multiplication is : %d\n", iResult);
    
    return 0;
}

File: Assignment__20/Question4.java
//Write a program to find the largest digit in a given number.

import java.util.Scanner;

class Function
{
    int DisplayLargeDigit(int iNo)
    {
        int iDigit = 0;
        int iMaxDigit = 0;

        while(iNo != 0)
        {
            iDigit = iNo % 10;
            if(iDigit > iMaxDigit)
            {
                iMaxDigit = iDigit;
            }
            iNo = iNo / 10; 
        }

        return iMaxDigit;
    }
}

class Question4
{
    public static void main(String A[])
    {
        Scanner sobj = new Scanner(System.in);

        System.out.println("Enter the number: ");
        int iValue = sobj.nextInt();

        int iResult = 0;
        
        Function fobj = new Function();
        iResult = fobj.DisplayLargeDigit(iValue);

        System.out.println("The largest digit is : " + iResult);

    }

}
